src/service/resources-impl.js

```diff
@@ -281,19 +284,61 @@ export class ResourcesImpl {
 
   /** @override */
   isIntersectionExperimentOn() {
-    return !!this.intersectionObserver_;
+    return !!this.oneViewportObserver_;
   }
 
   /**
    * @param {!Array<!IntersectionObserverEntry>} entries
-   * @param {!IntersectionObserver} unusedObserver
    * @private
    */
-  intersects_(entries, unusedObserver) {
-    dev().fine(TAG_, 'intersect', entries);
+  intersectsOneViewport_(entries) {
+    // TODO(willchou): Remove assert once #27167 is fixed.
+    devAssert(this.prerenderSize_ == 1);
 
-    const toUnload = [];
+    const whenBuilt = this.intersects_(entries, /* viewports */ 1);
 
+    // To prioritize in-viewport elements, delay handling of 2vp/3vp until
+    // all 1vp elements are built (and we're not in prerender, of course).
+    Promise.all([whenBuilt, this.ampdoc.whenFirstVisible()]).then(() => {
+      if (this.threeViewportsObserver_) {
+        return;
+      }
+      this.threeViewportsObserver_ = new IntersectionObserver(
+        e => this.intersectsThreeViewports_(e),
+        {
+          root: this.oneViewportObserver_.root,
+          rootMargin: '200% 25%',
```

Comment:
why is the margin width 25%?

Reply:
That's the current size of our loading rectangle (`loadRect` I think). One example is animating elements that start just off-screen, horizontally.

Reply:
Shouldn't we also have a `rootMargin: '0% 25%'` for the `oneViewportObserver_ ` then?

Reply:
`loadRect` in prerender mode doesn't have it. Also I think we'd want to deprioritize such elements relative to 1vp.

Reply:
ah, i see now. makes sense!